### PR TITLE
IBX-1116: Redesign Calendar: move react dropdown to common components

### PR DESCRIPTION
**Pull request by lucasOsti**
| Question      | Answer
| ------------- | ---
| Tickets       | https://issues.ibexa.co/browse/IBX-1116
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | no
| Doc needed?   | ?
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
 
-import { createCssClassNames } from '../../../common/helpers/css.class.names';
-import Icon from '../../../common/icon/icon';
+import { createCssClassNames } from '../../common/helpers/css.class.names';
+import Icon from '../../common/icon/icon';
 
-import { DropdownPortalRefContext } from '../../universal.discovery.module';
-
-const Dropdown = ({ value, options, onChange, small }) => {
+const Dropdown = ({ dropdownListRef, value, options, onChange, small, single, extraClass }) => {
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js:8]: single and extraClass are not in propTypes

File: src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js [hunk 2]
```diff
@@ -93,6 +91,10 @@ const Dropdown = ({ value, options, onChange, small }) => {
         )
     }
 
+    if (extraClass) {
+        dropdownClasses[extraClass] = true
```
Comment (dew326) [inline src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js:95]:
```suggestion
        dropdownClasses[extraClass] = true;
```

Comment (dew326) [reply src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js:95]: Why this cannot be in createCssClassNames with `[extraClass]: !!extraClass`?

**Review by dew326 — CHANGES_REQUESTED** [review]

File: src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js (hunk 2 above)
Comment (tischsoic) [reply src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js:95]: * `[extraClass]: true` with defaultProps `extraClasses: ''`.

File: src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js [hunk 3]
```diff
@@ -115,9 +116,11 @@ const ContentCreateWidget = () => {
                     <div className="ibexa-extra-actions__section-header">{selectLanguageLabel}</div>
                     <div class="ibexa-extra-actions__section-content">
                         <Dropdown
+                            dropdownListRef={DropdownPortalRefContext}
```
Comment (GrabowskiM) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js:119]: Is it correct? In Dropdown component as I see you want to get ref element (from useRef hook) and here you pass context

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
+const Dropdown = ({ dropdownListRef, value, options, onChange, small, single, extraClass }) => {
     const containerRef = useRef();
     const containerItemsRef = useRef();
-    const dropdownPortalRef = useContext(DropdownPortalRefContext);
     const [isExpanded, setIsExpanded] = useState(false);
     const [filterText, setFilterText] = useState('');
     const labelValue = options.find((option) => option.value === value)?.label;
-    const dropdownClassName = createCssClassNames({
-        'c-udw-dropdown ibexa-dropdown ibexa-dropdown--single': true,
+    const dropdownClasses = {
```
Comment (tischsoic) [inline src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js:14]:
```suggestion
    const className = createCssClassNames({
```

File: src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js (hunk 3 above)
Comment (tischsoic) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js:119]: It is not a ref, but a context of a ref
```suggestion
                            portalRefContext={DropdownPortalRefContext}
```

File: src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js [hunk 5]
```diff
@@ -115,9 +116,11 @@ const ContentCreateWidget = () => {
...
                             onChange={updateSelectedLanguage}
                             value={selectedLanguage}
                             options={languageOptions}
+                            extraClass="c-udw-dropdown"
```
Comment (tischsoic) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js:123]:
```suggestion
                            extraClasses="c-udw-dropdown"
```

File: src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js [hunk 6]
```diff
@@ -1,23 +1,21 @@
...
+import { createCssClassNames } from '../../common/helpers/css.class.names';
+import Icon from '../../common/icon/icon';
 
-import { DropdownPortalRefContext } from '../../universal.discovery.module';
-
-const Dropdown = ({ value, options, onChange, small }) => {
+const Dropdown = ({ dropdownListRef, value, options, onChange, small, single, extraClass }) => {
     const containerRef = useRef();
     const containerItemsRef = useRef();
-    const dropdownPortalRef = useContext(DropdownPortalRefContext);
```
Comment (tischsoic) [inline src/bundle/ui-dev/src/modules/common/dropdown/dropdown.js:13]: No need to `useContext` now? I'm not sure, just asking.
```
const dropdownPortalRef = useContext(dropdownListRef);
```

**Review by tischsoic — CHANGES_REQUESTED** [review]

File: src/bundle/ui-dev/src/modules/universal-discovery/components/filters/filters.js [hunk 7]
```diff
@@ -170,6 +170,7 @@ const Filters = ({ search }) => {
             <ContentTypeSelector />
             <Collapsible title={sectionLabel}>
                 <Dropdown
+                    dropdownListRef={DropdownPortalRefContext}
```
Comment (GrabowskiM) [inline src/bundle/ui-dev/src/modules/universal-discovery/components/filters/filters.js:173]: use still pass here Context instead of ref

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

**Review by GrabowskiM — APPROVED** [review]

**Review by dew326 — APPROVED** [review]

**Review by tischsoic — APPROVED** [review]

**Pull request merged**